upath/core.py

```diff
@@ -392,10 +425,19 @@ def owner(self):
     def readlink(self):
         raise NotImplementedError
 
-    def touch(self, truncate=True, **kwargs):
+    def touch(self, *args: int, truncate: bool = True, **kwargs) -> None:
+        if len(args) > 2:
+            raise TypeError("too many arguments")
+        else:
+            for key, val in zip(["mode", "exists_ok"], args):
+                if key in kwargs:
+                    raise TypeError(f"provided {key!r} as arg and kwarg")
+                kwargs[key] = val
```

Comment:
I don't understand why this is needed. Can you please explain in a comment?

Reply:
This keeps the calling signature compatible with Path so that mypy doesnt complain.

Reply:
done 👍